### PR TITLE
Added BRD_BOOT_DELAY parameter

### PR DESCRIPTION
**Pull request by tridge**
This allows for slower boots if the user has peripherals that need time to settle before being probed

### CODE REVIEW

File: libraries/AP_RangeFinder/RangeFinder.cpp [hunk 1]
```diff
@@ -380,6 +380,8 @@ void RangeFinder::detect_instance(uint8_t instance, uint8_t& serial_instance)
         break;
     case RangeFinder_TYPE_LWI2C:
         if (params[instance].address) {
+            // the LW20 needs a long time to boot up, so we delay 1.5s here
+            hal.scheduler->delay(1500);
```
Comment (WickedShell) [inline libraries/AP_RangeFinder/RangeFinder.cpp]: Presumably this should be bypassed if it's a watchdog reboot? Just noting that we are gaining lots of ways to skip fast boots for the watchdog scenario.

Comment (tridge) [reply libraries/AP_RangeFinder/RangeFinder.cpp]: fixed, thanks!

**Comment by magicrub** [comment]
Seems that what we really need to check for is a POR event.

**Comment by tridge** [comment]
> Seems that what we really need to check for is a POR event.

not for this PR. This PR is to mitigate issues caused by the code we've added recently that disables and then re-enables power on the peripheral bus while booting (added at the request of Hex to solve a different problem). So we know what time the power to that bus came on, and we just need to wait long enough for all peripherals to boot up sufficiently that we can probe for them

**Pull request merged**